src/Input.jsx

```diff
@@ -0,0 +1,67 @@
+import React, { PropTypes } from 'react';
+import classNames from 'classnames';
+
+const propTypes = {
+  children: PropTypes.node,
+  type: PropTypes.string,
+  size: PropTypes.string,
+  state: PropTypes.string,
+  tag: PropTypes.string,
+  'static': PropTypes.bool,
+  className: PropTypes.string,
+};
+
+const defaultProps = {
+  tag: 'p',
+  type: 'text',
+};
+
+const Form = (props) => {
+  const {
+    className,
+    type,
+    size,
+    state,
+    tag,
+    'static': staticInput,
+    ...attributes,
+  } = props;
+
+  const checkInput = ['radio', 'checkbox'].indexOf(type) > -1;
+
+  const fileInput = type === 'file';
+  const textareaInput = type === 'textarea';
+  const selectInput = type === 'select';
+  let Tag = selectInput || textareaInput ? type : 'input';
+
+  let formControlClass = 'form-control';
+
+  if (staticInput) {
+    formControlClass = `${formControlClass}-static`;
+    Tag = tag;
+  } else if (fileInput) {
+    formControlClass = `${formControlClass}-file`;
+  } else if (checkInput) {
+    formControlClass = 'form-check-input';
+  }
+
+  const classes = classNames(
+    className,
+    state ? `form-control-${state}` : false,
+    size ? `form-control-${size}` : false,
+    formControlClass
+  );
+
+  if (!staticInput && !selectInput) {
```

Comment:
if `textarea` and `select` stay within input, this should also check for textareaInput as to not add type to rendered dom.